pkg/virt-api/webhooks/validating-webhook/validating-webhook.go

```diff
@@ -1894,3 +1899,33 @@ func ValidateDuplicateDHCPPrivateOptions(PrivateOptions []v1.DHCPPrivateOptions)
 	}
 	return nil
 }
+
+func validateVMINodeNameLabel(oldVMI *v1.VirtualMachineInstance, newVMI *v1.VirtualMachineInstance) *metav1.StatusCause {
+	if oldVMI.ObjectMeta.Labels == nil && newVMI.ObjectMeta.Labels != nil {
```

Comment:
It is legitimate by our controller to add that label. At some point it is not there. I think it is important that we take into account who adds/changes the labels. The only user who is  allowed to modify `kubevirt.io/*` labels is the `virt-controller` user (a dedicated service account). You should be able to see the user in the review request.

Reply:
I've assumed (and was wrong, should've checked :smiley: ) that our components are not communicating through the API server. Added a fix for that.